packages/fxa-auth-server/lib/routes/session.js

```diff
@@ -325,81 +285,75 @@ module.exports = function(
       },
       handler: async function(request) {
         log.begin('Session.duplicate', request);
+
         const origSessionToken = request.auth.credentials;
 
-        return P.resolve()
-          .then(duplicateVerificationState)
-          .then(createSessionToken)
-          .then(formatResponse);
-
-        function duplicateVerificationState() {
-          // Copy verification state of the token, but generate
-          // independent verification codes.
-          const newVerificationState = {};
-          if (origSessionToken.tokenVerificationId) {
-            newVerificationState.tokenVerificationId = random.hex(
-              origSessionToken.tokenVerificationId.length / 2
-            );
-          }
-          if (origSessionToken.tokenVerificationCode) {
-            // Using expiresAt=0 here prevents the new token from being verified via email code.
-            // That's OK, because we don't send them a new email with the new verification code
-            // unless they explicitly ask us to resend it, and resend only handles email links
-            // rather than email codes.
-            newVerificationState.tokenVerificationCode = random.hex(
-              origSessionToken.tokenVerificationCode.length / 2
-            );
-            newVerificationState.tokenVerificationCodeExpiresAt = 0;
-          }
-          return P.props(newVerificationState);
+        // Copy verification state of the token, but generate
+        // independent verification codes.
+        let newVerificationState = {};
+        if (origSessionToken.tokenVerificationId) {
+          newVerificationState.tokenVerificationId = random.hex(
```

Comment:
I assume these return promises? That's why there's `P.props` below? If so, why not await them here and avoid the `P.props` entirely?

Reply:
Yep. Just for efficiency I think, so they're awaited concurrently.